handlers/download.go

```diff
@@ -147,3 +154,32 @@ func (d Download) Do(extension string) http.HandlerFunc {
 		http.Error(w, notFoundMessage, http.StatusNotFound)
 	}
 }
+
+func (d Download) authenticate(r *http.Request, logData map[string]interface{}) (bool, map[string]interface{}) {
+	var authorised bool
+
+	if d.IsPublishing {
+		var hasCallerIdentity, hasUserIdentity bool
+
+		callerIdentity := identity.Caller(r.Context())
+		if callerIdentity != "" {
+			logData["caller_identity"] = callerIdentity
+			hasCallerIdentity = true
+		}
+
+		userIdentity := identity.User(r.Context())
```

Comment:
I don't think you need to check for a user identity here. The caller identity will contain either the calling service or calling user. The user identity can be populated from an incoming request header so is not safe to use to determine authentication. See the `identity.isPresent(ctx)` function in go-ns, as it does what you need in this function : https://github.com/ONSdigital/go-ns/blob/b7c32727ee0de2d1e987539302b141890d287b6a/identity/authentication.go#L35